File: infrastructure/src/main/java/com/example/app/movie/repository/MovieRepository.java
package com.example.app.movie.repository;

import com.example.app.movie.entity.Movie;
import com.example.app.movie.type.MovieStatus;
import org.springframework.data.domain.Sort;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.List;

public interface MovieRepository extends JpaRepository<Movie, Long> {
    List<Movie> findAllByReleaseDateLessThanEqualAndStatus(LocalDate showDate, MovieStatus status, Sort sort);
}

File: src/main/kotlin/com/example/demo/domain/repository/MovieRepository.kt
package com.example.demo.domain.repository

import com.example.demo.domain.entity.Movie
import org.springframework.data.jpa.repository.JpaRepository

interface MovieRepository : JpaRepository<Movie, Long> {
    fun findByTitleContainingIgnoreCaseAndGenreContainingIgnoreCase(
        title: String?,
        genre: String?
    ): List<Movie>
} 